### PR TITLE
added few sorting algorithms

### PR DESCRIPTION
**Pull request by Pratiknarola**

### CODE REVIEW

File: pygorithm/sorting/cocktail_sort.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+Cocktail Sort traverses through a given array in both directions alternatively.
+
+'''
+
+def cocktailSort(a): 
+    n = len(a) 
+    swapped = True
+    start = 0
+    end = n-1
+    while (swapped == True):
```
Comment (Tjstretchalot) [inline pygorithm/sorting/cocktail_sort.py]:
```py
while swapped:
```

is sufficient here

File: pygorithm/sorting/cocktail_sort.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    while (swapped == True): 
+  
+        # reset the swapped flag on entering the loop, 
+        # because it might be true from a previous 
+        # iteration. 
+        swapped = False
+  
+        # loop from left to right same as the bubble 
+        # sort 
+        for i in range (start, end):
```
Comment (Tjstretchalot) [inline pygorithm/sorting/cocktail_sort.py]: Can remove the space after "range": https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements

File: pygorithm/sorting/cocktail_sort.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  
+        # reset the swapped flag on entering the loop, 
+        # because it might be true from a previous 
+        # iteration. 
+        swapped = False
+  
+        # loop from left to right same as the bubble 
+        # sort 
+        for i in range (start, end): 
+            if (a[i] > a[i + 1]) :
```
Comment (Tjstretchalot) [inline pygorithm/sorting/cocktail_sort.py]: Would be better without parenthesis or space before colon:

```py
if a[i] > a[i + 1]:
```

File: pygorithm/sorting/cocktail_sort.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  
+        # loop from left to right same as the bubble 
+        # sort 
+        for i in range (start, end): 
+            if (a[i] > a[i + 1]) : 
+                a[i], a[i + 1]= a[i + 1], a[i] 
+                swapped = True
+  
+        # if nothing moved, then array is sorted. 
+        if (swapped == False):
```
Comment (Tjstretchalot) [inline pygorithm/sorting/cocktail_sort.py]:
```py
if not swapped:
```

is equivalent and more pythonic

File: pygorithm/sorting/cocktail_sort.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        if (swapped == False): 
+            break
+  
+        # otherwise, reset the swapped flag so that it 
+        # can be used in the next stage 
+        swapped = False
+  
+        # move the end point back by one, because 
+        # item at the end is in its rightful spot 
+        end = end-1
```
Comment (Tjstretchalot) [inline pygorithm/sorting/cocktail_sort.py]: spaces between operators is the pep8 guideline, but in this case there's a combined operator:

```
end -= 1
```

File: pygorithm/sorting/cocktail_sort.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  
+        # from right to left, doing the same 
+        # comparison as in the previous stage 
+        for i in range(end-1, start-1, -1): 
+            if (a[i] > a[i + 1]):
```
Comment (Tjstretchalot) [inline pygorithm/sorting/cocktail_sort.py]: the parenthesis is not necessary here

```py
if a[i] > a[i + 1]:
```

File: pygorithm/sorting/cocktail_sort.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        # reset the swapped flag on entering the loop, 
+        # because it might be true from a previous 
+        # iteration. 
+        swapped = False
+  
+        # loop from left to right same as the bubble 
+        # sort 
+        for i in range (start, end): 
+            if (a[i] > a[i + 1]) : 
+                a[i], a[i + 1]= a[i + 1], a[i]
```
Comment (Tjstretchalot) [inline pygorithm/sorting/cocktail_sort.py]: missing a space:

```py
a[i], a[i + 1] = a[i + 1], a[i]
```

**Comment by Pratiknarola** [comment]
I have changed the cocktail_sort.py as suggested. And thanks for the advice. Eventually, these small tips help to write better code. If there is still any scope for improvement, please let me know.

**Comment by Tjstretchalot** [comment]
Thank you for the changes! Could you rename the functions to be snake_case? (i.e. timSort -> tim_sort)

Looking at the sorting unit tests they aren't the best right now, otherwise I would suggest adding those. However, I can refactor the sorting unit tests to be more comprehensive and less boilerplate first

**Comment by Pratiknarola** [comment]
@OmkarPathak Thanks for the advice. I have changed the function names as suggested. If there are any changes possible, please let me know.

File: pygorithm/sorting/tim_sort.py [hunk 8]
```diff
@@ -0,0 +1,68 @@
+# iterative Timsort function to sort the  
+# array[0...n-1] (similar to merge sort)  
+def tim_sort(arr, n):  
+   
+    # Sort individual subarrays of size RUN  
+    for i in range(0, n, RUN):  
+        insertionSort(arr, i, min((i+31), (n-1)))
```
Comment (OmkarPathak) [inline pygorithm/sorting/tim_sort.py]: Where is insertion sort defined? This would throw an error

Comment (OmkarPathak) [reply pygorithm/sorting/tim_sort.py]: You can import it from same folder using: `from insertion_sort import sort`

**Comment by Pratiknarola** [comment]
@OmkarPathak Sorry for that. I forgot to do that and Sorry for so many mistakes. I have checked all files and I guess, now this should be okay but still if there is any changes, please let me know. Thanks.

**Pull request merged**